plotly/tools.py

```diff
@@ -1451,6 +1455,823 @@ class FigureFactory(object):
+
+        import numpy as np
+        import pandas as pd
+        from scipy import stats
+
+        # create dataframe
+        np.random.seed(619517)
+        Nr=250
+        y = np.random.randn(Nr)
+        gr = np.random.choice(list("ABCDE"), Nr)
+        norm_params=[(0, 1.2), (0.7, 1), (-0.5, 1.4), (0.3, 1), (0.8, 0.9)]
+
+        for i, letter in enumerate("ABCDE"):
+            y[gr == letter] *=norm_params[i][1]+ norm_params[i][0]
+        df = pd.DataFrame(dict(Score=y, Group=gr))
+
+        # define header params
+        data_header = 'Score'
+        group_header = 'Group'
+
+        # make groupby object with pandas
+        groupby_data = df.groupby([group_header])
+
+        for group in "ABCDE":
+            data_from_group = groupby_data.get_group(group)[data_header]
+            # take a stat of the grouped data
+            stat = np.median(data_from_group)
+            # add to dictionary
+            group_stats[group] = stat
+
+        # create violin fig
+        fig = FF.create_violin(df, data_header='Score', group_header='Group',
+                               height=600, width=1000, use_colorscale=True,
+                              group_stats=group_stats)
+
+        # plot
+        py.iplot(fig, filename='Violin Plot with Colorscale')
+        ```
+        """
+        if _numpy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "numpy to be imported.")
+
+        if _scipy_imported is False:
+            raise ImportError("FigureFactory.create_violin() requires "
+                              "scipy to be imported.")
+
+        from plotly.graph_objs import graph_objs
+        from numbers import Number
+        plotly_scales = {'Greys': ['rgb(0,0,0)', 'rgb(255,255,255)'],
+                         'YlGnBu': ['rgb(8,29,88)', 'rgb(255,255,217)'],
+                         'Greens': ['rgb(0,68,27)', 'rgb(247,252,245)'],
+                         'YlOrRd': ['rgb(128,0,38)', 'rgb(255,255,204)'],
+                         'Bluered': ['rgb(0,0,255)', 'rgb(255,0,0)'],
+                         'RdBu': ['rgb(5,10,172)', 'rgb(178,10,28)'],
+                         'Reds': ['rgb(220,220,220)', 'rgb(178,10,28)'],
+                         'Blues': ['rgb(5,10,172)', 'rgb(220,220,220)'],
+                         'Picnic': ['rgb(0,0,255)', 'rgb(255,0,0)'],
+                         'Rainbow': ['rgb(150,0,90)', 'rgb(255,0,0)'],
+                         'Portland': ['rgb(12,51,131)', 'rgb(217,30,30)'],
+                         'Jet': ['rgb(0,0,131)', 'rgb(128,0,0)'],
+                         'Hot': ['rgb(0,0,0)', 'rgb(255,255,255)'],
+                         'Blackbody': ['rgb(0,0,0)', 'rgb(160,200,255)'],
+                         'Earth': ['rgb(0,0,130)', 'rgb(255,255,255)'],
+                         'Electric': ['rgb(0,0,0)', 'rgb(255,250,220)'],
+                         'Viridis': ['rgb(68,1,84)', 'rgb(253,231,37)']}
```

Comment:
Hmm, did we discuss moving this to a constant? Seems weird to create this. Maybe we should create a `colors.py` file for things like this?